plugins/woocommerce/includes/rest-api/Controllers/Version3/class-wc-rest-products-controller.php

```diff
@@ -323,6 +323,24 @@ protected function prepare_objects_query( $request ) {
 		return $args;
 	}
 
+	/**
+	 * Get object.
+	 *
+	 * @param int $id Object ID.
+	 *
+	 * @since  3.0.0
+	 * @return WC_Data
+	 */
+	protected function get_object( $id ) {
+		/**
+		 * Filter the product object that is fetched in the context of the REST API.
+		 *
+		 * @param WC_Product $product The product.
+		 * @since 9.0.0
+		 */
+		return apply_filters( 'woocommerce_rest_get_product', wc_get_product( $id ) );
```

Comment:
> We would have to run the code from possibly_add_template_id and also take care of adding the updated meta_data to the response.

Technically we run this anyhow right?

One thought, instead of actually saving the product the product with the updated meta data right away we could only modify the response data. This way once a user saves the product within the new form for the first time the template id gets added, otherwise it doesn't.

Reply:
> I preferred adding a new filter because, when woocommerce_rest_prepare_product_object runs

I am still not a big fan of adding a filter, also partially because this "in theory" is a temporary need, only to support backwards compatibility with the old editor.
I did again prefer to make use of what is already available.
Maybe others within @woocommerce/mothra have an opinion on this also?

Reply:
I don't really love that we are modifying the product in the DB when we are getting the product. That sort of flow feels confusing and I worry about potential inadvertent loops or hooks being called when not expected.